src/Interpreters/PartLog.h

```diff
@@ -18,6 +18,11 @@ struct PartLogElement
         MOVE_PART = 6,
     };
 
+    static constexpr size_t max_query_id_size = 128;
```

Comment:
Maybe simply use std::string?

(the query id in CurrentThread is plain array only for the purpose to be able to obtain it from signal handlers)

Reply:
Ok, I just see this way is used in here, so what is the purpose here?
  https://github.com/ClickHouse/ClickHouse/blob/8b98465f10340c6c41caee05a1bed428b316b2e2/src/Storages/System/StorageSystemStackTrace.cpp#L55-L57

Reply:
Also for signal safety (StorageSystemStackTrace is passing data from signal handler).